packages/rich-text/src/component/use-anchor.js

```diff
@@ -162,7 +169,19 @@ export function useAnchor( { editableContentElement, settings = {} } ) {
 				getAnchor( editableContentElement, tagName, className )
 			);
 		}
-	}, [ editableContentElement, tagName, className, isActive, wasActive ] );
+	}, [
+		editableContentElement,
+		tagName,
+		className,
+		isActive,
+		wasActive,
+		// Active attributes is defaulted to a stable object to avoid re-rendering,
```

Comment:
It's unclear what the "stable object" idea should offer.

Reply:
I agree. See

> The Link UI position gets "stuck" on the second clicked link

...from [this comment](https://github.com/WordPress/gutenberg/pull/59635#issuecomment-1983462373).

I am loath to modify the API of `useAnchor` (even if it's "safe") but without this "fix" this PR will not working because the anchor will not be recomputed.

We've run into this issue before but somehow our UI masked the error but it keeps surfacing.

Reply:
I am actually wondering if I made the function signature use a object is that object "stable" (by which I mean, it is not a new _reference_ on each call of the hook).

The reason why I want a stable object, is because it is one of the deps to the `useLayoutEffect` and thus if it changes on each render then the effect will run more often then is necessary. For the purposes of backwards compatibility I want to make sure there are no performance regressions introduced by providing this new setting option.


```
const {
	tagName,
	className,
	isActive,
	__unstableActiveAttributes: activeAttributes = {}, // just a standard object!
} = settings;
```